site/public/js/ta-grading-rubric.js

```diff
@@ -1832,11 +1832,30 @@ function onClickComponent(me) {
  * @param me DOM Element of the cancel button
  */
 function onCancelComponent(me) {
-    toggleComponent(getComponentIdFromDOMElement(me), false)
-        .catch(function (err) {
+  const component_id = getComponentIdFromDOMElement(me);
+  const gradeable_id = getGradeableId();
+  const anon_id = getAnonId();
+  ajaxGetGradedComponent(gradeable_id, component_id, anon_id).then((component)=>{
+    // If there is any changes made in comment of a component , prompt the TA
+    if ( component.comment !== $('#component-' + component_id).find('.mark-note-custom').val().trim()) {
```

Comment:
I don't think trimming the message is required in this case since adding/removing whitespace could be a valid change the user would like to discard.

Reply:
done!